### PR TITLE
simulator: add TiltCase and Main

### PR DESCRIPTION
**Pull request by nolouch**
ref https://github.com/pingcap/pd/issues/853

### CODE REVIEW

File: pkg/faketikv/client.go [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+	reportRegionHeartbeatCh chan *core.RegionInfo
+	reciveRegionHeartbeatCh chan *pdpb.RegionHeartbeatResponse
+
+	wg     sync.WaitGroup
+	ctx    context.Context
+	cancel context.CancelFunc
+}
+
+// NewClient creates a PD client.
+func NewClient(pdAddrs []string) (Client, <-chan *pdpb.RegionHeartbeatResponse, error) {
```
Comment (disksing) [inline pkg/faketikv/client.go]: I think we can simplify it by not supporting multiple pd-servers.

**Comment by nolouch** [comment]
PTAL

File: pkg/faketikv/initializer.go [hunk 2]
```diff
@@ -13,20 +13,129 @@
...
+		id, err1 := c.alloc.AllocID()
+		if err1 != nil {
+			log.Fatal("alloc failed", err1)
+		}
+		node, err1 := NewNode(id, fmt.Sprintf("mock://tikv-%d", id), addr)
+		if err != nil {
+			log.Fatal("New node failed", err1)
+		}
+		nodes[id] = node
+		if len(ids) < 3 {
```
Comment (Connor1996) [inline pkg/faketikv/initializer.go]: why?

Comment (nolouch) [reply pkg/faketikv/initializer.go]: I use the ids to generate region, all regions will distribute in the three nodes.

File: pkg/faketikv/initializer.go [hunk 3]
```diff
@@ -13,20 +13,129 @@
...
+		peer := &metapb.Peer{
+			Id:      id,
+			StoreId: storeID,
+		}
+		peers = append(peers, peer)
+	}
+	regionMeta := &metapb.Region{
+		Id:          regionID,
+		StartKey:    []byte(fmt.Sprintf("zt_%d", start)),
+		EndKey:      []byte(fmt.Sprintf("zt_%d", start+1000)),
```
Comment (disksing) [inline pkg/faketikv/initializer.go]: '%d' cannot guarantee order, use something like '%20d' instead.

**Comment by disksing** [comment]
LGTM. PTAL @Connor1996

**Comment by siddontang** [comment]
please add the build command into the Makefile

**Comment by siddontang** [comment]
seem it can't work now? @nolouch

**Comment by nolouch** [comment]
@siddontang  it can work.

File: Makefile [hunk 4]
```diff
@@ -71,4 +71,7 @@ endif
 	mkdir -p _vendor
 	mv vendor _vendor/src
 
+simulator:
```
Comment (Connor1996) [inline Makefile]: it can't by triggered by make dev, ci will not check it. Should we add it in dev command?

Comment (nolouch) [reply Makefile]: yep, PTAL

**Pull request merged**